### PR TITLE
Test ezid sending link and changing/deleting link on a volume

### PR DESCRIPTION
**Pull request by kanishka-azimi**
## Description
see title

## Motivation and Context
prepare for changing query implementation, expand bdd tests

## How Has This Been Tested?
this is a test

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] I have run the unit tests.
- [x] I am sick of writing content for this PR, let's get this over with.

### CODE REVIEW

File: test/ModelTest.hs [hunk 1]
```diff
@@ -476,6 +476,29 @@ test_15b = Test.stepsWithTransaction "test_15b" $ \step cn2 -> do
...
+    link <- Gen.sample genVolumeLink
+    (v, [loadedLink]) <- runReaderT
+        (do
+            v <- addVolumeWithAccess aiAcct
+            changeVolumeLinks v [link]
+            ls <- lookupVolumeLinks v
+            pure (v, ls))
+        aiCtxt
+    step "When the link is changed and removed"
+    step "Then one can see each change"
```
Comment (chreekat) [inline test/ModelTest.hs]: `step` out of place

File: test/ModelTest.hs [hunk 2]
```diff
@@ -476,6 +476,29 @@ test_15b = Test.stepsWithTransaction "test_15b" $ \step cn2 -> do
     volLinks <- runWithNoIdent cn2 (lookupVolumeLinks v)
     volLinks @?= [link]
 
+test_15c :: TestTree
+test_15c = Test.stepsWithTransaction "test_15c" $ \step cn2 -> do
```
Comment (chreekat) [inline test/ModelTest.hs]: `Test.stepsWithTransaction "change volume links"` maybe

File: test/ModelTest.hs [hunk 3]
```diff
@@ -507,14 +530,23 @@ test_17 :: TestTree
 test_17 = localOption (mkTimeout (10 * 10^(6 :: Int))) $ Test.stepsWithResourceAndTransaction "test_17" $ \step ist cn2 -> do
```
Comment (chreekat) [inline test/ModelTest.hs]: "ezid generation - volume with link" maybe

File: test/ModelTest.hs (hunk 1 above)
Comment (kanishka-azimi) [reply test/ModelTest.hs]: intentional, I wasn't sure how to interleave. considered creating two tests, but felt like too much test noise.

File: test/ModelTest.hs (hunk 3 above)
Comment (kanishka-azimi) [reply test/ModelTest.hs]: aim of this is ezid generation - volume with all data. It only has a link currently

**Pull request merged**